### PR TITLE
Adds chat notification when all citizens (other than guards) are sleeping

### LINKED ISSUE #5068: Send notification when all citizens are in bed.
### Minecolonies version # Update to the newest minecolonies and structurize alpha first and see if your issue might be already fixed!

Send a notification when all citizens are in bed. This is particularly useful so you can know when to sleep but not have your citizens complain about not getting enough sleep. 

Version: 0.11.948-ALPHA

### Expected behavior

- When all citizens are in bed send a notification to colony officers stating that all citizens are in bed. 

### Actual behaviour

- No notifications are sent

### Steps to reproduce the problem

1. 
2. 
3. 
4. ... Profit

### Logs: Latest.log and crashlog at https://gist.github.com/

- latest.log:
- crashlog:

### PR DESCRIPTION
**Pull request by PowerSchill**
Closes #5068
Closes #
Closes #

# Changes proposed in this pull request:
- Adds chat notification when all citizens (other than guards) are sleeping so that the player knows when it is ok for them to sleep.
- 
-

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/5070/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java [hunk 1]
```diff
@@ -548,4 +550,31 @@ public ICitizenData getRandomCitizen()
...
+    {
+        this.isCitizensSleeping = sleep;
+    }
+
+    @Override
+    public void onCitizenSleep()
+    {
+
+
+        for (final ICitizenData citizenData : citizens.values())
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:567]: This is actually more then needed. We have already a way to check on the sleep handler if a citizen is sleeping. So you only need to check this, no need for additional variables.

Comment (PowerSchill) [reply src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:567]: `this.isCitizensSleeping` is used to mark that all citizens are sleeping, not just a particular citizen.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:567]: I mean, that is not necessary. You can just, when a citizen lies down, check if all other citizen are sleeping already. And if so, print the message.

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java [hunk 2]
```diff
@@ -548,4 +550,31 @@ public ICitizenData getRandomCitizen()
...
+        {
+            if (!(citizenData.isAsleep() || citizenData.getJob() instanceof AbstractJobGuard))
+            {
+                return;
+            }
+        }
+
+        if (!this.isCitizensSleeping)
+        {
+            LanguageHandler.sendPlayersMessage(colony.getImportantMessageEntityPlayers(), ALL_CITIZENS_ARE_SLEEPING);
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java]: getImportantMessageEntityPlayers will notify any officers even when not inside the colony, e.g. your friend who is officer in your colony aswell. Think you want to use the nonimportant ones

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java (hunk 1 above)
Comment (PowerSchill) [reply src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:567]: Oh, I remember now. The citizen, for some reason, calls `trySleep` multiple times even after they sleep which lead to several notifications being triggered. This was my way to reset the notifications until the next day.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:567]: Then there seems to be code missing in the PR, does this compile?

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java [hunk 3]
```diff
@@ -82,6 +84,11 @@
      */
     private Random random = new Random();
 
+    /**
+     * Whether all citizens excluding guards are sleeping
+     */
+    private boolean isCitizensSleeping;
+
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:90]: areCitizensSleeping then

**Comment by Raycoms** [comment]
Also please resolve conflict

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:90]: Still this here =D

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java [hunk 4]
```diff
@@ -548,4 +554,31 @@ public ICitizenData getRandomCitizen()
...
+    @Override
+    public void updateCitizenSleep(final boolean sleep)
+    {
+        this.areCitizensSleeping = sleep;
+    }
+
+    @Override
+    public void onCitizenSleep()
+    {
+
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java]: Just this 2 lines of whitespace here and then we can merge it =D

**Pull request merged**